packages/fxa-content-server/app/tests/spec/views/pair/auth_allow.js

```diff
@@ -116,7 +123,56 @@ describe('views/pair/auth_allow', () => {
         });
     });
 
-    it('blocks users with TOTP', () => {
+    it('handles users with TOTP', () => {
+      account.accountProfile.restore();
+      sinon.stub(account, 'accountProfile').callsFake(() => {
+        return Promise.resolve({
+          authenticationMethods: ['pwd', 'email', 'otp'],
+          authenticatorAssuranceLevel: 1,
+          email: MOCK_EMAIL,
+        });
+      });
+      sinon.spy(view, 'replaceCurrentPage');
+      return view.render()
+        .then(() => {
+          assert.isTrue(view.replaceCurrentPage.calledOnceWith('pair/auth/totp'));
+        });
+    });
+
+    it('handles users after TOTP with existing token', () => {
+      model.set('totpComplete', true);
+      sinon.stub(account, 'checkTotpTokenExists').callsFake(() => {
+        return Promise.resolve({
+          exists: true,
+        });
+      });
+
+      account.accountProfile.restore();
+      sinon.stub(account, 'accountProfile').callsFake(() => {
+        return Promise.resolve({
+          authenticationMethods: ['pwd', 'email', 'otp'],
+          authenticatorAssuranceLevel: 1,
+          email: MOCK_EMAIL,
+        });
+      });
+      sinon.spy(view, 'replaceCurrentPage');
+      return view.render()
+        .then(() => {
+          $('#container').html(view.el);
+          assert.isFalse(view.replaceCurrentPage.calledOnceWith('pair/auth/totp'));
+          assert.isTrue(view.$el.find('#fxa-pair-auth-allow-header').text().includes(MOCK_EMAIL));
+          assert.equal(view.$el.find('.family-os').text(), 'Firefox on Windows');
+          assert.equal(view.$el.find('.location').text().trim(), 'Toronto, Ontario, Canada (estimated)');
+          assert.equal(view.$el.find('.ip-address').text(), 'IP address: 1.1.1.1');
+        });
+    });
+
+    it('handles users after TOTP with existing token throwing an error', () => {
+      model.set('totpComplete', true);
+      sinon.stub(account, 'checkTotpTokenExists').callsFake(() => {
+        return Promise.reject(new Error('Unverified Session'));
```

Comment:
Hrm, so I guess this is the case that you're trying to handle that I commented on above, throwing "unverified session" after completing TOTP.  This seems quite concerning - how did we manage to complete TOTP without the session becoming verified? Have you observed this happening in the wild, after fixing the "we were using the wrong sessionToken" problem?

Reply:
Yeah I had mixed feelings about this. I think we can drop this check for now, it doesn't add anything useful for users or the totp flow itself.

> Have you observed this happening in the wild

No we fixed that now.